frontend/talentsearch/src/js/components/applicantProfile/ExperienceSection.tsx

```diff
@@ -0,0 +1,233 @@
+import { Tab, TabSet } from "@common/components/tabs";
+import {
+  BookOpenIcon,
+  BriefcaseIcon,
+  LightBulbIcon,
+  StarIcon,
+  UserGroupIcon,
+} from "@heroicons/react/solid";
+import * as React from "react";
+import { useIntl } from "react-intl";
+import { EducationExperience } from "@common/api/generated";
+import { getLocale } from "@common/helpers/localize";
+import {
+  isAwardExperience,
+  isCommunityExperience,
+  isEducationExperience,
+  isPersonalExperience,
+  isWorkExperience,
+} from "../../types/ExperienceUtils";
+import {
+  AwardExperience,
+  CommunityExperience,
+  Experience,
+  PersonalExperience,
+  Skill,
+  WorkExperience,
+} from "../../api/generated";
+import ExperienceAccordion from "../ExperienceAccordion/ExperienceAccordion";
+import SkillAccordion from "../skills/SkillAccordion/SkillAccordion";
+
+export type ExperienceForDate =
+  | (AwardExperience & { startDate: string; endDate: string })
+  | CommunityExperience
+  | EducationExperience
+  | PersonalExperience
+  | WorkExperience;
+
+export const compareByDate = (e1: ExperienceForDate, e2: ExperienceForDate) => {
+  const e1EndDate = e1.endDate ? new Date(e1.endDate).getTime() : null;
+  const e2EndDate = e2.endDate ? new Date(e2.endDate).getTime() : null;
+  const e1StartDate = e1.startDate ? new Date(e1.startDate).getTime() : -1;
+  const e2StartDate = e2.startDate ? new Date(e2.startDate).getTime() : -1;
+
+  // All items with no end date should be at the top and sorted by most recent start date.
+  if (!e1EndDate && !e2EndDate) {
+    return e2StartDate - e1StartDate;
+  }
+
+  if (!e1EndDate) {
+    return -1;
+  }
+
+  if (!e2EndDate) {
+    return 1;
+  }
+
+  // Items with end date should be sorted by most recent end date at top.
+  return e2EndDate - e1EndDate;
+};
+const ExperienceByType: React.FunctionComponent<{
+  title: string;
+  icon: React.ReactNode;
+  experiences: Experience[];
+}> = ({ title, icon, experiences }) => {
+  return (
+    <div>
+      <div data-h2-display="b(flex)" data-h2-margin="b(top-bottom, m)">
+        {icon}
+        <p
+          data-h2-font-size="b(h4)"
+          data-h2-margin="b(all, none)"
+          data-h2-padding="b(left, s)"
+        >
+          {title}
+        </p>
+      </div>
+      <div
+        data-h2-radius="b(s)"
+        data-h2-bg-color="b(lightgray)"
+        data-h2-padding="b(top-bottom, xxs) b(right-left, xs)"
+      >
+        {experiences.map((experience) => (
+          <ExperienceAccordion key={experience.id} experience={experience} />
+        ))}
+      </div>
+    </div>
+  );
+};
+export interface ExperienceSectionProps {
+  experiences?: Experience[];
+}
+
+const ExperienceSection: React.FunctionComponent<ExperienceSectionProps> = ({
+  experiences,
+}) => {
+  const intl = useIntl();
+  const locale = getLocale(intl);
+
+  const awardExperiences =
+    experiences
+      ?.filter((experience) => isAwardExperience(experience))
```

Comment:
While the ExperienceSection is being refactored anyway, I can recommend a tiny stylistic change. 😁 Passing functions directly into `filter()`.
```suggestion
      ?.filter(isAwardExperience)
```